hw/templates/snax_accelerator_top_wrapper.sv.tpl

```diff
@@ -0,0 +1,275 @@
+    //-------------------------------
+    // Clocks and reset
+    //-------------------------------
+    .clk_i                ( clk_i                 ),
+    .rst_ni               ( rst_ni                ),
+    //-----------------------------
+    // CSR control ports
+    //-----------------------------
+    // Request
+    .csr_req_addr_i       ( acc_csr_req_addr [0]  ),
+    .csr_req_data_i       ( acc_csr_req_data [0]  ),
+    .csr_req_write_i      ( acc_csr_req_wen  [0]  ),
+    .csr_req_valid_i      ( acc_csr_req_valid[0]  ),
+    .csr_req_ready_o      ( acc_csr_req_ready[0]  ),
+
+    // Response
+    .csr_rsp_data_o       ( acc_csr_rsp_data [0]  ),
+    .csr_rsp_ready_i      ( acc_csr_rsp_valid[0]  ),
+    .csr_rsp_valid_o      ( acc_csr_rsp_ready[0]  ),
+
+    //-----------------------------
+    // Packed CSR register signals
+    //-----------------------------
+    .csr_reg_set_o        ( acc_csr_reg_set       ),
+    .csr_reg_set_valid_o  ( acc_csr_reg_set_valid ),
+    .csr_reg_set_ready_i  ( acc_csr_reg_set_ready )
+  );
+
+  //-----------------------------
+  // Accelerator
+  //-----------------------------
+  // Note: This is the part that needs to be consistent
+  // It needs to have the correct connections to the control and data ports!
+
+  ${cfg["tag_name"]}_wrapper #(
+    .NumInputPorts        ( NumInputPorts         ),
+    .NumOutputPorts       ( NumOutputPorts        ),
+    .DataWidth            ( NarrowDataWidth       )
+  ) i_${cfg["tag_name"]}_wrapper (
+    //-------------------------------
+    // Clocks and reset
+    //-------------------------------
+    .clk_i                ( clk_i                 ),
+    .rst_ni               ( rst_ni                ),
+
+    //-----------------------------
+    // Accelerator ports
+    //-----------------------------
+    // Input ports from streamer to accelerator
+    .stream2acc_data_i    ( stream2acc_data       ),
+    .stream2acc_valid_i   ( stream2acc_valid      ),
+    .stream2acc_ready_o   ( stream2acc_ready      ),
+
+    // Output ports from accelerator to streamer
+    .acc2stream_data_o    ( acc2stream_data       ),
+    .acc2stream_valid_o   ( acc2stream_valid      ),
+    .acc2stream_ready_i   ( acc2stream_ready      ),
+
+    //-----------------------------
+    // Packed CSR register signals
+    //-----------------------------
+    .csr_reg_set_i        ( acc_csr_reg_set       ),
+    .csr_reg_set_valid_i  ( acc_csr_reg_set_valid ),
+    .csr_reg_set_ready_o  ( acc_csr_reg_set_ready )
+  );
+
+  //-----------------------------
+  // Streamer Wrapper
+  //-----------------------------
+  ${cfg["tag_name"]}_streamer_wrapper #(
+    .NarrowDataWidth          ( NarrowDataWidth         ),
+    .TCDMDepth                ( TCDMDepth               ),
+    .TCDMReqPorts             ( TCDMReqPorts            ),
+    .TCDMSize                 ( TCDMSize                ),
+    .TCDMAddrWidth            ( TCDMAddrWidth           )
+  ) i_streamer_wrapper (
```

Comment:
Maybe changing `i_streamer_wrapper` to `i_${cfg["tag_name"]}_streamer_wrapper` is better? Otherwise if there are multiple accelerators, this `i_streamer_wrapper` module will cause multiple definition error